src/crystal/system/win32/time.cr

```diff
@@ -11,14 +11,6 @@ module Crystal::System::Time
   NANOSECONDS_PER_SECOND    = 1_000_000_000
   FILETIME_TICKS_PER_SECOND = NANOSECONDS_PER_SECOND / NANOSECONDS_PER_FILETIME_TICK
 
-  # TODO: For now, this method returns the UTC offset currently in place, ignoring *seconds*.
```

Comment:
Since time currently works on windows, can we not break it on windows? Having a working win32 implementation of this would make me confident that the `Crystal::System::Time` abstraction you've chosen works on windows.

Reply:
Would it be ok with you if I assured you that I have a more-or-less working implementation for windows waiting? It's just missing conversion from `WCHAR*` to `String` for `Location.load_local` and `File` implementation for `Location.load`.
For obvious reasons, `Crystal::System::Time.zone_sources` will be empty on Windows because the OS doesn't include a copy of the tzdatabase.

Reply:
so how will zone sources work on windows?

Reply:
At least in Go they seem to ship with a `zoneinfo.zip` in the installation, and check if that exists as a last resource: https://github.com/golang/go/blob/master/lib/time/zoneinfo.zip

Reply:
AFAIK TimeZone information is available in Windows in the registry (`HKLM\SOFTWARE\Microsoft\Windows NT\CurrentVersion\Time Zones`). Not sure how that can be made accessible to Crystal instead of `zoneinfo.zip` mechanism, or if that is relevant as mechanism to retrieve timezone information on Windows vs non-Windows platforms.

Cheers.

Reply:
This should be done natively. Problem is that that registry data is probably a different format, which is why I bought up this issue.

Reply:
Getting useful timezone data in Windows is a nasty problem. That's enough headeache to fuel another discussion. Basically: it's shit and you can't do anything about it.

Essentialy it comes down to two options: You either distribute a copy of the timezone database with your application or you can't use locations except UTC and local.

@luislavena You could try to get something out of Windows' time zone registry but that wont get you far. Windows just doesn't provide complete information. It contains only the standard and dst (if available) offset and dates applicable for the current year. That's not even worth trying, considering that the names are completely different from the IANA database. That would require a translation table to be able to use the same location names for windows and unix. Then it's much easier and better to just provide a copy of the tzdatabase. It's about 350 KB, so no big deal regarding distribution size. The main issue is that it's your responsibility to see it get updated.

@asterite That is not the last ressort but as far as I can tell the *only* way except manually providing a path to a database via `ZONEINFO` envvar. Otherwise it only works if you have Go installed (see go/golang#14276). - which will obviously not be the case on all machines running applications written in Go.

Reply:
So if I understand it correctly, there would be no good Time::Location support out of the box for Windows, and it's your responsibility to distribute that zip file with your application. I guess that's fine with me. We can improve this stuff later, but the only other option would be to somehow embed this data in the application, but that's probably way too much data.

Reply:
Well, we should provide a way to optionally embed the zip data inside the executable on windows, but I'll accept your point that it's impossible.